src/multisig/multisigTransactionsOutcomeParser.ts

```diff
@@ -0,0 +1,278 @@
+import { TransactionOnNetwork } from "../core";
+import { Address } from "../core/address";
+import { Err } from "../core/errors";
+import { TransactionEvent } from "../core/transactionEvents";
+import { SmartContractCallOutcome, SmartContractResult } from "../transactionsOutcomeParsers/resources";
+
+enum Events {
+    MultisigDeploy = "MultisigDeploy",
+    MultisigActionProp = "MultisigActionProposal",
+    SignalError = "signalError",
+    WriteLog = "writeLog",
+}
+
+/**
+ * Parses the outcome of multisig contract operations
+ */
+export class MultisigTransactionsOutcomeParser {
+    /**
+     * Parses the outcome of creating a new multisig contract
+     * @param transactionOnNetwork The completed transaction
+     * @returns An array of objects containing the new contract addresses
+     */
+    parseDeployMultisigContract(transactionOnNetwork: TransactionOnNetwork): { contractAddress: Address }[] {
+        const directCallOutcome = this.findDirectMultisigDeployOutcome(transactionOnNetwork);
+
+        if (!directCallOutcome || directCallOutcome.returnCode !== "ok") {
+            return [];
+        }
+
+        // Look for the deployment events
+        const events = transactionOnNetwork.logs.events
+            .concat(transactionOnNetwork.smartContractResults.flatMap((result) => result.logs.events))
+            .filter((event) => event.identifier === Events.MultisigDeploy);
```

Comment:
There are some utility functions here:
 - https://github.com/multiversx/mx-sdk-js-core/blob/main/src/transactionsOutcomeParsers/resources.ts

Perhaps `findEventsByIdentifier` is helpful.